include/fi.h

```diff
@@ -60,6 +60,15 @@ extern "C" {
 #endif
 
 /*
+ * Windows doesn't have __BYTE_ORDER, add it manually
+ */
+#ifdef _WIN32
+#define LITTLE_ENDIAN 5678
+#define BIG_ENDIAN 8765
+#define BYTE_ORDER LITTLE_ENDIAN
+#endif
+
+/*
```

Comment:
Please add a new directory, include/windows.  Move all windows specific definitions into include/windows/osd.h.  There should be almost no ifdef code anywhere.